javascript/ql/test/query-tests/Security/CWE-079/exception-xss.js

```diff
@@ -174,4 +174,13 @@ app.get('/user/:id', function(req, res) {
 	new Promise(resolve => inner(foo, resolve)).catch((e) => {
       $('myId').html(e); // NOT OK
     });
-})();
\ No newline at end of file
+})();
+
+app.get('/user/:id', function(req, res) {
+  unknown(req.params.id, (error, res) => {
+    if (error) {
+	  $('myId').html(error); // OK (falls through to the next statement) 
```

Comment:
hmm, looking at this source code makes me less convinced that we must have the reachability information we are fighting with above. I can imagine programs like:

```
(req, res) => {
  if (error) { 
    msg = error; 
    type ='error'; 
  } 
  ...; 
  res.send(msg) 
}
```
Lets make a note to try this without the reachability information later, and  see which new `error` parameters we identify. We can also try it out now if you like.

Reply:
This is already not a high precision query, so we can reintroduce the reachability information if we get too many false positives. 

I'll remove it for now.